### PR TITLE
Okta-380150 Add SSR and SIW requirements for Registration Hook Guide and API Reference

### PR DESCRIPTION
**Pull request by brianduffield-okta**
## Description:
**What's changed?** 
- Added a note in the [Registration Inline Hook API ](https://developer.okta.com/docs/reference/registration-hook/) reference page to highlight SSR must be enabled and used with SIW. Also fixed a broken code block.
- Added same note and info on SSR to [Registration Inline Hook Guide ](https://developer.okta.com/docs/guides/registration-inline-hook/nodejs/overview/)

**Is this PR related to a Monolith release?** No

### Resolves:

* [OKTA-380150](https://oktainc.atlassian.net/browse/OKTA-380150)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/guides/registration-inline-hook/overview/index.md
Comment (jakubvul-okta) [inline packages/@okta/vuepress-site/docs/guides/registration-inline-hook/overview/index.md:7]:
```suggestion
You must enable [self-service registration (SSR)](/docs/guides/set-up-self-service-registration/before-you-begin/) to implement a Registration Inline Hook.
```

**Review by jakubvul-okta — APPROVED** [review]
One change, otherwise 🚢

**Comment by brianduffield-okta** [comment]
Waiting for final tech sign-off.

**Comment by brianduffield-okta** [comment]
Received sign-off.

**Pull request merged**
